### DISCUSSION TITLE
How BLOCKS configuration works?

### DISCUSSION THREAD

**Discussion by vijayank88** [The-OpenROAD-Project/OpenROAD-flow-scripts#984] [Q&A] [answered]
How this `BLOCKS` variable working? https://github.com/The-OpenROAD-Project/OpenROAD-flow-scripts/blob/19ae71b8646ffc287042942f70d82a9ecba1b6f8/flow/designs/asap7/uart-blocks/config.mk#L9

Is only one instance can be used or multiple instance can be used with `BLOCKS` variable?

**Comment by maliberty** [comment, marked as answer]
See https://github.com/The-OpenROAD-Project/OpenROAD-flow-scripts/blob/19ae71b8646ffc287042942f70d82a9ecba1b6f8/flow/Makefile#L186

where we generate additional targets.   Also https://github.com/The-OpenROAD-Project/OpenROAD-flow-scripts/blob/19ae71b8646ffc287042942f70d82a9ecba1b6f8/flow/Makefile#L317

You can have multiple blocks.

**Comment by vijayank88** [comment]
I was thinking to automate `pdn_grid_strategy` for macro grids.
For memory macros the existing macro grid works fine. 
```
define_pdn_grid -name {CORE_macro_grid_1} -voltage_domains {CORE} -macro -orient {R0 R180 MX MY} -cells {.*}
```
For `BLOCKS` it is not connecting properly. 
For example if we use right cell name `define_pdn_grid -macro -cells uart_rx ` the grid generates properly.

**Reply by maliberty** [reply]
Is your instance in some other orientation?  .* should match all cells.
